x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/EvaluateDataFrameAction.java

```diff
@@ -0,0 +1,207 @@
+            }
+            this.indices = indices.toArray(new String[indices.size()]);
+        }
+
+        public Evaluation getEvaluation() {
+            return evaluation;
+        }
+
+        public final void setEvaluation(Evaluation evaluation) {
+            this.evaluation = ExceptionsHelper.requireNonNull(evaluation, EVALUATION);
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            super.readFrom(in);
+            indices = in.readStringArray();
+            evaluation = in.readNamedWriteable(Evaluation.class);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeStringArray(indices);
+            out.writeNamedWriteable(evaluation);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.array(INDEX.getPreferredName(), indices);
+            builder.startObject(EVALUATION.getPreferredName());
+            builder.field(evaluation.getName(), evaluation);
+            builder.endObject();
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(Arrays.hashCode(indices), evaluation);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request that = (Request) o;
+            return Arrays.equals(indices, that.indices) && Objects.equals(evaluation, that.evaluation);
+        }
+    }
+
+    static class RequestBuilder extends ActionRequestBuilder<Request, Response> {
+
+        RequestBuilder(ElasticsearchClient client) {
+            super(client, INSTANCE, new Request());
+        }
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private EvaluationResult result;
+
+        public Response() {
+        }
+
+        public Response(EvaluationResult result) {
+            this.result = Objects.requireNonNull(result);
+        }
+
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            super.readFrom(in);
+            this.result = in.readNamedWriteable(EvaluationResult.class);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeNamedWriteable(result);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field(result.getName(), result);
```

Comment:
This allows the response to have the evaluation type at the top level.

The response for `binary_soft_classification` currently is (for example): 

```
{
    "binary_soft_classification": {
        "auc_roc": {
            "score": 0.7680318097975196
        }
    }
}
```

If I took that away it would become:

```
{
  "auc_roc": {
      "score": 0.7680318097975196
  }
}
```

The problem with that would be that a client would not know how to parse this as different evaluations would result in different responses.